### PR TITLE
Clean up tests that trigger the race detector

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Added some mux test hooks and supporting test infrastructure.

Fixes https://github.com/Azure/go-amqp/issues/169

### CODE REVIEW

File: sender.go [hunk 1]
```diff
@@ -340,6 +340,7 @@ Loop:
 		case tr := <-outgoingTransfers:
 			select {
 			case s.l.session.txTransfer <- &tr:
+				// TODO: this logging races with assignment of DeliveryID by Session.mux()
```
Comment (jhendrixMSFT) [inline sender.go:343]: Will fix in a separate PR.

Comment (jhendrixMSFT) [reply sender.go:343]: I will also fix CI so that the race detector actually runs :(

Comment (jhendrixMSFT) [reply sender.go]: OK I lied and fixed it now along with enabling the race detector.

**Review by richardpark-msft — APPROVED** [review]
Love the hooks!

**Pull request merged**